test/conftest.py

```diff
@@ -0,0 +1,133 @@
+'''
+Utility functions used by the cibuildwheel tests.
+'''
+
+import subprocess, sys, os
+import pytest
+
+
+IS_WINDOWS_RUNNING_ON_AZURE = os.path.exists('C:\\hostedtoolcache')
+IS_WINDOWS_RUNNING_ON_TRAVIS = os.environ.get('TRAVIS_OS_NAME') == 'windows'
+
+platform = None
+
+if 'CIBW_PLATFORM' in os.environ:
+    platform = os.environ['CIBW_PLATFORM']
+elif sys.platform.startswith('linux'):
+    platform = 'linux'
+elif sys.platform.startswith('darwin'):
+    platform = 'macos'
+elif sys.platform in ['win32', 'cygwin']:
+    platform = 'windows'
+else:
+    raise Exception('Unsupported platform')
+
+
+@pytest.fixture
+def utils(tmp_path):
+    '''
+    Fixture use to run cibuildwheel with outputdir in a tmp_path, also list the wheels in that tmp_path
+    '''
+    class _utils:
+
+        def cibuildwheel_get_build_identifiers(self, project_path, env=None):
+            '''
+            Returns the list of build identifiers that cibuildwheel will try to build
+            for the current platform.
+            '''
+            cmd_output = subprocess.check_output(
+                [sys.executable, '-m', 'cibuildwheel', '--print-build-identifiers', project_path],
+                universal_newlines=True,
+                env=env,
+            )
+
+            return cmd_output.strip().split('\n')
+
+        def cibuildwheel_run(self, project_path, env=None, add_env=None):
+            '''
+            Runs cibuildwheel as a subprocess, building the project at project_path.
+
+            Uses the current Python interpreter.
+            Configure settings using env.
+            '''
+            if env is None:
+                env = os.environ.copy()
+
+            if add_env is not None:
+                env.update(add_env)
+
+            subprocess.check_call(
+                [sys.executable, '-m', 'cibuildwheel', '--output-dir', str(tmp_path), project_path],
+                env=env,
+            )
```

Comment:
If this parts are moved to fixture then maybe eliminate need of project_path:
```python
@pytest.fixture
def utils(tmp_path, request):
    '''
    Fixture use to run cibuildwheel with outputdir in a tmp_path, also list the wheels in that tmp_path
    '''
    class _utils:

        def cibuildwheel_get_build_identifiers(self, project_path=None, env=None):
            '''
            Returns the list of build identifiers that cibuildwheel will try to build
            for the current platform.
            '''

            if project_path is None:
                project_path = getattr(request.module, "PROJECT_PATH", os.path.dirname(getattr(request.module, "__file__")))

            cmd_output = subprocess.check_output(
                [sys.executable, '-m', 'cibuildwheel', '--print-build-identifiers', project_path],
                universal_newlines=True,
                env=env,
            )

            return cmd_output.strip().split('\n')

        def cibuildwheel_run(self, project_path=None, env=None, add_env=None):
            '''
            Runs cibuildwheel as a subprocess, building the project at project_path.

            Uses the current Python interpreter.
            Configure settings using env.
            '''
            if project_path is None:
                project_path = getattr(request.module, "PROJECT_PATH", os.path.dirname(getattr(request.module, "__file__")))

            if env is None:
                env = os.environ.copy()

            if add_env is not None:
                env.update(add_env)

            subprocess.check_call(
                [sys.executable, '-m', 'cibuildwheel', '--output-dir', str(tmp_path), project_path],
                env=env,
            )
```
then tests could be simplified 
```python
import os
import sys 


def test(utils):
    # build the wheels
    utils.cibuildwheel_run()

    # check that the expected wheels are produced
    expected_wheels = utils.expected_wheels('spam', '0.1.0')
    actual_wheels = utils.list_wheels()
    assert set(actual_wheels) == set(expected_wheels)


def test_build_identifiers(utils):
    # check that the number of expected wheels matches the number of build
    # identifiers
    # after adding CIBW_MANYLINUX_IMAGE to support manylinux2010, there
    # can be multiple wheels for each wheel, though, so we need to limit
    # the expected wheels
    expected_wheels = [w for w in utils.expected_wheels('spam', '0.1.0')
                       if not '-manylinux' in w or '-manylinux1' in w]
    build_identifiers = utils.cibuildwheel_get_build_identifiers()
    assert len(expected_wheels) == len(build_identifiers)
```

Reply:
Yessss, nice! I knew there should be a way to get rid of that repeating `PROJECT_DIR` :-)

I would argue to make it even simpler, though:

```
if project_path is None:
    project_path = os.path.dirname(request.module.__file__")
```

Or even get rid of `project_path` as parameter. If we need it again for some specific reason in the future, it's easy enough to add again.

```
project_path = os.path.dirname(request.module.__file__")
```

Reply:
> Or even get rid of project_path as parameter. If we need it again for some specific reason in the future, it's easy enough to add again.

> `project_path = os.path.dirname(request.module.__file__)`

No possibility of give `project_path` may be significant when writing new test (like test for c++ standards). It is easier to write separated project utils and then integrate it to existed one.

Reply:
This now done. I kept the possibility to override `project_path`.